Modify S5147(Common text): Update to the education framework (APPSEC-94) (#1195)
SonarSource/rspec: rules/S5147/common/fix/dangerous-operators.adoc

==== Operators are to be classified as dangerous

Because these operators run JavaScript, restrict the use of dangerous operators
and functions in both untrusted input and internal code, such as:

* `$where`
* `$function`
* `$accumulator`
* `mapReduce`

> Depending on your use case, you should first try all the ways to use regular

Reviewer: ```suggestion
Depending on your use case, you should first try using regular
```

API calls before using any of these operators. +

> For example, a `$where` operator is excessive when only a simple search is

Author: ```suggestion
For example, using a `$where` operator is unnecessarily complex when only a simple search is
```

> For example, using a `$where` operator is unnecessarily complex when only a simple search is
> required. It also leads to performance problems.

Author: ```suggestion
For example, using a `$where` operator is unnecessarily complex when only a
simple search is required. It also leads to performance problems.
```

**Note**: https://www.mongodb.com/docs/manual/reference/operator/query/where/#javascript-enablement[Server-side scripting can be disabled].

> Regular operators can also lead to data leaks. +

Reviewer: Seems it requires further description.
